### PR TITLE
Extend VisibilityConfig to private methods

### PR DESCRIPTION
**Pull request by pongad**
This commit extends VisibilityConfig to include private methods. The main motivation of these private methods is to let us create hand-written methods in their places. The private methods can be used by the hand-written ones to simplify work.

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/go_main_library.baseline [hunk 1]
```diff
@@ -606,6 +611,22 @@ func (c *Client) FindRelatedBooks(ctx context.Context, req *librarypb.FindRelate
...
+// addLabel adds a label to the entity.
+func (c *Client) addLabel(ctx context.Context, req *taggerpb.AddLabelRequest) (*taggerpb.AddLabelResponse, error) {
+    md, _ := metadata.FromContext(ctx)
+    ctx = metadata.NewContext(ctx, metadata.Join(md, c.metadata))
+    var resp *taggerpb.AddLabelResponse
+    err := gax.Invoke(ctx, func(ctx context.Context) error {
+        var err error
+        resp, err = c.labelerClient.addLabel(ctx, req)
+        return err
+    }, c.CallOptions.AddLabel...)
```
Comment (jba) [inline src/test/java/com/google/api/codegen/testdata/go_main_library.baseline]: I guess we have to make the CallOptions field public even though the method is private. Fine for now. We may have to revisit someday.

File: src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java [hunk 2]
```diff
@@ -149,6 +149,21 @@ public MethodTransformerContext asMethodContext(Method method) {
     return methods;
   }
 
+  /** Returns a list of methods with samples, similar to getSupportedMethods,
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java]: If you have a multi-line comment, I don't think the text is supposed to start on the first line

File: src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java [hunk 3]
```diff
@@ -149,6 +149,21 @@ public MethodTransformerContext asMethodContext(Method method) {
...
+   *  but also filter out private methods.
+   */
+  public List<Method> getMethodsWithSamples() {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java]: It seems like "getPublicMethods" would be better - there is no check in the logic for having samples.

File: src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java (hunk 2 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java]: Done.

File: src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java (hunk 3 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java]: Good idea, done.

**Pull request merged**